### ISSUE TITLE
Where is the createapp command documented?

### ISSUE THREAD

**Issue by metaperl** [reahl/reahl#142]
After you complete the tutorial and want to create a new app, there is no info on how to do so:

1. the [tutorial ](https://www.reahl.org/docs/4.0/tutorial/index.d.html)itself never mentions how - it shows how to deploy an app but not how to create one
2. [the documention](https://www.reahl.org/docs/4.0/index.d.html) doesnt say how
3. `reahl help-commands` does not list any such command:

I suppose for now I will just take an example similar to mine and then modify it.

```bash
$ reahl help-commands
usage: reahl [-h] [-l {CRITICAL,ERROR,WARNING,INFO,DEBUG}] command ...

Invoke reahl commands on the commandline.

positional arguments:
  command               a command
  command_args

optional arguments:
  -h, --help            show this help message and exit
  -l {CRITICAL,ERROR,WARNING,INFO,DEBUG}, --loglevel {CRITICAL,ERROR,WARNING,INFO,DEBUG}
                        set log level to this

"reahl help-commands" gives a list of available commands


Commands: ( reahl <command> --help for usage on a specific command)

addlocale            Adds a new locale catalog.
alias                Adds an alias.
backupall            Backs up all the databases on the host this project config
                     points to.
backupdb             Backs up the database.
build                Builds all distributable packages for each project in the
                     current selection.
checkconfig          Checks current configuration settings.
clearselection       Clears the current selection.
compiletranslations  Compiles all current translations.
createdb             Creates the database.
createdbtables       Creates all necessary tables in the database.
createdbuser         Creates the database user.
debianise            Debianises a project.
debinstall           Runs setup.py install with correct arguments for packaging
                     the result in a deb (for each project in the selection).
delete               Deletes the saved selection set with the given name.
diffdb               Prints out a diff between the current database schema and
                     what is expected by the current code.
dropdb               Drops the database.
dropdbtables         Drops all necessary tables in the database.
dropdbuser           Drops the database user.
example              Give you a copy of an example.
explainlegend        Prints an explanation of project status indicators.
extractmessages      Collects strings marked for translation.
info                 Prints information about a project.
list                 Lists all the projects currently selected (or, optionally,
v                     all projects in the workspace)
listconfig           Lists current configuration settings.
listdeps             List all dependency eggs in dependency order.
listexamples         Lists available examples.
markreleased         Marks each project in the current selection as released.
mergetranslations    Merges newly discovered strings needing translation with a
                     catalog of existing translations.
migratedb            Runs all necessary database migrations.
missingdeps          Lists all missing thirdparty dependencies for each project
                     in the current selection.
read                 Changes the current selection to the named, previously
                     saved selection.
refresh              Reconstructs the working set of projects by searching
                     through the given directories (by default the entire
                     workspace directory).
restoreall           Restores all the databases on the host this project config
                     points to.
restoredb            Restores the database.
runjobs              Runs all registered scripts.
save                 Saves the current selection of projects using the given
                     name.
select               Selects a subset of projects in the workspace based on
                     their state.
selections           Lists all the named sets of selections that have been saved
                     previously.
serve                Serves the project configured in the given directory
                     (defaults to ./etc).
servesmtp            Runs an SMTP server on port 8025 for testing.
setup                Runs setup.py <command> for each project in the current
                     selection.
shell                Executes a shell command in each selected project, from
                     each project's own root directory.
sizedb               Prints the current size of the database.
substvars            Generates debian substvars.
syncfiles            Uses the sitecopy program to upload / sync static files to
                     a remote webserver.
unalias              Removes an alias.
update-copyright     Updates the dates found in copyright statements found on
                     the first line of files based on git history.
updateapt            Updates the index files of the given apt repository.
upload               Uploads all built distributable packages for each project
                     in the current selection.

No Aliasses
```

**Comment by IwanVosloo** [comment]
There is no "createapp" command as is done by projects such as django. The reason simply being that creating an app in reahl does not require so much boilerplate code and directory structure.

That said, we can easily add one. I can see it is handy and moreover, familiar!

Good suggestion, I think we should add one.

**Comment by metaperl** [comment]
In my experience, pretty much all you need to do is modify one of the examples But perhaps the app name would be different. So something like `reahl example-as` or `reahl clone-example` would be more useful to me immediately.

Concretely speaking, my current app is based off `datatable bootstrap` and I went through and changed the word `Address` everywhere to the name more appropriate for my domain. But I left the main file as `datatablebootstrap.py` because for a long time I could not find the file to specify the root of the app.

In other words, using an example as a template is probably preferred by most people. Because there are quite a few files to manually create yourself if you try from scratch.And you have to get the syntax right.

1. .realhproject
1. web.config.py
1. the main app file

**Comment by metaperl** [comment]
Yes, creation of a reahl app is non-trivial. It would be taxing to create the proper files needed:
* .reahlalias
* .reahlproject
* etc\web.config.py
* prod\etc\web.config.py
* prod\etc\reahl.config.py
* prod\etc\systemaccountmodel.py

But the examples are not consistent and complete in their generation of a complete set of sample files. For instance `reahl example datatablebootstrap`  does not create the `prod/*` hiearchy whereas `helloanywhere` does.

**Comment by metaperl** [comment]
Also I think the standard format should have `dev` and `prod` directories sibling to each other. Right now there is an `etc` directory which really should be a level down under a `dev` directory.

**Comment by metaperl** [comment]
Also the word `etc` is imprecise. This is a directory of config files.... Why do we need an `etc` directory at all? Why not just `dev` and `prod`?

**Comment by IwanVosloo** [comment]
Good points, thanks. I think we should definitely then add a createapp command that sets up the basics, perhaps by asking a few questions as to what certain things should be named.

Cloneexample is trickier to do generically since each example is focussed on just what is is explaining without other fluff. 

createapp could also ask stuff like "Do you want to an app that's internationalised? and add the relevant stuff only if you want it"

**Comment by IwanVosloo** [comment]
etc is a convention that comes from unix, which means in the unix world "the directory in which you save config files".

Most examples do not have production configuration, since that's not what they are examples of. The helloanywhere example IS an example of deploying something to prod, that's why it has prod config.

I would want to stick to keeping examples as simple as possible, and focused on what they explain only.

For a createapp command though it makes sense to have a more complete skeleton.

Perhaps we should change everything to consistently have:
 etc\dev

and for things that need to be deployed, also:
 etc\prod

Which can then both be generated by a createapp command.
